music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -0,0 +1,437 @@
+"""SiriusXM music provider support for MusicAssistant."""
+from __future__ import annotations
+
+from collections import defaultdict
+from collections.abc import AsyncGenerator
+from typing import TYPE_CHECKING, Generator, Union, Dict, Final
+
+from sxm import SXMClientAsync
+from sxm.models import QualitySize, RegionChoice, XMCategory, XMChannel, XMLiveChannel
+
+from music_assistant.common.models.config_entries import ConfigEntry, ConfigValueType
+from music_assistant.common.models.enums import LinkType, ProviderFeature, ConfigEntryType, ContentType
+from music_assistant.common.models.media_items import (
+    AudioFormat,
+    BrowseFolder,
+    ImageType,
+    MediaItemImage,
+    MediaItemLink,
+    MediaItemType,
+    MediaType,
+    ProviderMapping,
+    Radio,
+    StreamDetails
+)
+from music_assistant.server.models.music_provider import MusicProvider
+
+SUPPORTED_FEATURES = (ProviderFeature.BROWSE, ProviderFeature.LIBRARY_RADIOS,)
+
+if TYPE_CHECKING:
+    from music_assistant.common.models.config_entries import ProviderConfig
+    from music_assistant.common.models.provider import ProviderManifest
+    from music_assistant.server import MusicAssistant
+    from music_assistant.server.models import ProviderInstanceType
+
+CONF_SXM_USERNAME: Final[str] = "sxm_email_address"
+CONF_SXM_PASSWORD: Final[str] = "sxm_password"
+CONF_SXM_REGION: Final[str] = "sxm_region"
+
+GENERATION_POP = 0
+GENERATION_DECADES = 1
+GENERATION_ELSE = 2
+
+DECADE_CATEGORY_KEY_RANGES = {
+    '50s60s': (1950, 1970),
+    '70s': (1970, 1980),
+    '80s': (1980, 1990),
+    '90s': (1990, 2000),
+    '00s': (2000, 2010),
+    '10s': (2010, 2020),
+    '20s': (2020, 2030),
+}
+
+
+async def setup(
+        mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    prov = SiriusXMProvider(mass, manifest, config)
+
+    await prov.handle_setup()
+    return prov
+
+
+# noinspection PyUnusedLocal

```

Comment:
none of our linters should require this ? what editor are you using for Python ?

Reply:
PyCharm Professional. It showed `mass`, `instance_id`, `action`, `values` as unused params w/warnings. I wasn't sure if I could just remove them or should just ignore.

Reply:
You may never remove them but instead you inform the linter that it may be ignored.
We use the linters that are configured in the pre-commit config

Reply:
We all use VSCode btw ;-)